vllm/v1/core/scheduler.py

```diff
@@ -310,6 +343,10 @@ def schedule(self) -> "SchedulerOutput":
                         self.encoder_cache_manager.allocate(request, i)
                     encoder_budget = new_encoder_budget
 
+        # Put back any skipped requests at the head of the waiting queue
+        if still_waiting:
+            self.waiting = still_waiting + self.waiting
```

Comment:
should we clear still_waiting here?

Reply:
still_waiting will be cleared per step() call, so we don't have to do anything 😃